### PR TITLE
(test): Run E2E Tests on Push and Pull Requests

### PR DESCRIPTION
**Pull request by Bcoderx6**
This PR introduces a new GitHub Actions workflow to run end-to-end (E2E) tests on every commit and pull request (PR) to the main branch. The new workflow is based on the existing E2E workflow used for releases but is modified to remove tag checkouts, ensuring that all tests are executed against the head of each monorepo @jayasanka-sack

### CODE REVIEW

File: .github/workflows/e2e-tests-on-commit.yml. [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Extract version numbers from the spa-build-config.json file
+        id: refs
+        run: bash e2e_test_support_files/extract_tag_numbers.sh
```
Comment (jayasanka-sack) [inline .github/workflows/e2e-tests-on-commit.yml.:20]: No need
```suggestion

```

File: .github/workflows/e2e-tests-on-commit.yml. [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+        run: |
+          gzip -d ${{ steps.tempdir.outputs.tmpdir }}/e2e_release_env_images.tar.gz
+          docker load --input ${{ steps.tempdir.outputs.tmpdir }}/e2e_release_env_images.tar
+          docker image ls -a
+
+      - name: Spin up an OpenMRS Instance
+        run: docker-compose up -d
+        working-directory: e2e_test_support_files
+
+      - name: Checkout to the Repo's Head
```
Comment (jayasanka-sack) [inline .github/workflows/e2e-tests-on-commit.yml.:64]:
```suggestion
      - name: Checkout to the repo's main branch
```

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Great work, @Bcoderx6! I’ve added a few minor comments for you to review.

**Comment by jayasanka-sack** [comment]
Also fix the filename: `.github/workflows/e2e-tests-on-commit.yml.` -> `.github/workflows/e2e-tests-on-commit.yml` (extra dot at the end)

File: .github/workflows/e2e-tests-on-commit.yml. [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
```
Comment (jayasanka-sack) [inline .github/workflows/e2e-tests-on-commit.yml.:17]:
```suggestion
```

File: .github/workflows/e2e-tests-on-commit.yml. (hunk 2 above)
Comment (jayasanka-sack) [reply .github/workflows/e2e-tests-on-commit.yml.]: @Bcoderx6 do the same for other jobs too.

File: .github/workflows/e2e-tests-on-commit.yml [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
```
Comment (jayasanka-sack) [inline .github/workflows/e2e-tests-on-commit.yml:17]:
```suggestion
```

File: .github/workflows/e2e-tests-on-commit.yml [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+
+      - name: Build and Run Containers
+        run: docker-compose -f e2e_test_support_files/docker-compose-build.yml up -d
```
Comment (jayasanka-sack) [inline .github/workflows/e2e-tests-on-commit.yml:20]:
```suggestion
    steps:
      - uses: actions/checkout@v4

      - name: Build and Run Containers
        run: docker-compose -f e2e_test_support_files/docker-compose-build.yml up -d
```

**Comment by Bcoderx6** [comment]
Thank you, @jayasanka-sack! I’ve fixed the issues you mentioned. Please have another look and let me know if there are any more changes needed.

**Review by jayasanka-sack — APPROVED** [review]
Thanks for getting this done @Bcoderx6 ! 👏

**Pull request merged**